mesonbuild/modules/sourceset.py

```diff
@@ -125,7 +177,7 @@ def add_all_method(self, state: ModuleState, args: T.Tuple[T.List[SourceSetImpl]
         for s in if_true:
             if not isinstance(s, SourceSetImpl):
                 raise InvalidCode('Arguments to \'add_all\' after the first must be source sets')
-            s.frozen = True
+        s.frozen = True
```

Comment:
This seems broken, we need to mark each sourceset as frozen once we add its contents into our new sourceset. After this change, we only mark the last one as frozen.